staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -336,11 +308,42 @@ func (a *Allocator) Allocate(ctx context.Context, node *v1.Node) (finalResult []
 			}
 		}
 		for _, config := range claim.Spec.Devices.Config {
-			allocationResult.Devices.Config = append(allocationResult.Devices.Config, resourceapi.DeviceAllocationConfiguration{
-				Source:              resourceapi.AllocationConfigSourceClaim,
-				Requests:            config.Requests,
-				DeviceConfiguration: config.DeviceConfiguration,
-			})
+			// If Requests are empty, it applies to all. So it can just be included.
```

Comment:
Do I understand correctly that this is this a behavior change? In 1.32, it always include all the configs, but since it has requests in there it leaves it up to the drivers to apply the correct ones to the correct devices? And now, we still rely on the drivers to do that, but we exclude any configs that are associated with a request that was not one of the chosen ones? In 1.32, where were *no* requests that were not part of the allocations, so I think this is equivalent, unless it was possible to add a config before with an invalidate request name. Those would be included in 1.32 and excluded now, IIUC. I think that's OK, if it's the case...

Reply:
Good point. With the implementation in this PR, we end up with a filtering of the configs in both the allocator and the drivers, which seems unnecessary. It is probably easier to make a clear decision that the filtering should happen in either the allocator or the drivers. Are there any benefits to doing this in the drivers? It seems like doing it once in the allocator rather than having every driver implement it seems like the best option.

Reply:
IMHO, best to do it in the allocator. @aojea also found the current behavior surprising while working on a driver.

Reply:
How do you propose to move that logic into the allocator? If the same config applies to multiple devices, would you change the ResourceClaimStatus so that for each device, you list all configs that apply by device, even if that means listing them multiple times?

Reply:
We could provide a helper function in `k8s.io/dynamic-resource-allocation/resourceclaim` which extracts all "relevant" configs for a specific device from the allocation status. Then device driver authors don't need to implement this themselves and we get the same effect as changing the API - at least for drivers written in Go.

What remains is the question whether extending the semantic of the request field is a breaking change. I think it is, because "old" drivers will not match configs correctly once the new `<request name>/<subrequest name>` format is involved. I think that's okay: it's an alpha feature, so there is time to roll out new drivers as needed.

Reply:
That sounds like a good plan. Better than changing the API. Thanks Patrick.

Reply:
Can we do the helpers in this PR? The test driver would be the first user of them. The corresponding E2E test would need to be sufficiently complex that the new semantic matters.